lesson16/app.js

```diff
@@ -1,187 +1,164 @@
-const getJsonUrl = "https://jsondata.okiba.me/v1/json/oxP76210607104654";
-const toDomUl = document.getElementById("js-parent");
+const parent = document.getElementById("js-parent");
 
-//カテゴリタブ
-const tabsUl = document.createElement("ul");
-tabsUl.classList.add("tabs");
-const tabsContainerLi = document.createElement("li");
-tabsContainerLi.classList.add("tabs_container");
-const tabsFragment = document.createDocumentFragment();
+const renderLoading = (pearent) => {
+  const loadingImage = document.createElement("span");
+  loadingImage.id = "js-loadingImage";
+  loadingImage.classList.add("loader");
+  pearent.appendChild(loadingImage);
+};
+const removeLoading = () => document.getElementById("js-loadingImage").remove();
 
-//カテゴリ記事
-const tabContentsUl = document.createElement("ul");
-tabContentsUl.classList.add("tab_contents");
-const tabContentsContainerLi = document.createElement("li");
-tabContentsContainerLi.classList.add("tab_contents_container");
-const contentsFragment = document.createDocumentFragment();
+const renderErrorMessage = (error) => {
+  const errorMessage = document.createElement("p");
+  errorMessage.textContent = `エラー内容：${error.message}`;
+  parent.appendChild(errorMessage);
+  console.error(error.message);
+};
 
-//カテゴリタブ・記事をDOMへ
-toDomUl.prepend(tabsContainerLi);
-toDomUl.appendChild(tabContentsContainerLi).appendChild(tabContentsUl);
-tabsContainerLi.appendChild(tabsUl);
-
-async function getJsonData() {
+const getData = async () => {
   try {
-    const response = await fetch(getJsonUrl);
-    const jsonData = await response.json();
-    return jsonData;
-  } catch (err) {
-    throw Error(err);
-  } finally {
-    console.log("終了しました");
+    const response = await fetch(
+      "https://mocki.io/v1/d173a984-04a5-4ebe-b66d-52ad4060d664"
+    );
+    if (response.ok) return await response.json();
+    throw new Error(`${response.status}:${response.text}`);
+  } catch (e) {
+    console.log(e);
+    throw new Error(e);
   }
-}
+};
 
-async function init() {
+const init = async () => {
+  renderLoading(parent);
   let data;
   try {
-    data = await getJsonData();
+    data = await getData();
   } catch (e) {
-    tabsUl.innerHTML = `error is ${e}`;
-  }
-  if (data.length !== 0) {
-    createElements(data);
-  } else {
-    tabsUl.innerHTML = "data is empty";
+    renderErrorMessage(e);
+    return;
+  } finally {
+    removeLoading();
   }
-}
+  if (data.data.length === 0) return (parent.textContent = "data is empty");
+  renderTabs(data);
```

Comment:
nits:
How about putting the following code in the "try" block to get the contents of the "throw" and using the "renderErrorMessage (e)" inside the "catch" block to display the error?

[This was taught by "yuka-san".](https://github.com/yasu-pro/js-handson/pull/12#discussion_r824488430)

```suggestion
   if (data.data.length === 0) throw new Error("data is empty");
   renderTabs(data);
```

Reply:
Thanks for your comment! Moriken-sensei commented earlier that you should not throw an error if the data is empty, so that's what I did.